From d9f337f8fc494531143c12a571cac2c54d0008bf Mon Sep 17 00:00:00 2001
From: Arnon Kaufman <akaufman@habana.ai>
Date: Sun, 14 Apr 2024 15:41:33 +0300
Subject: [PATCH] [SW-15774] build: add common paths files

Preparation for changing the build products location.
Centralize the different paths variables into one place

Change-Id: I1827615a430d80f52156fe871b13bcf9ee51e215
Signed-off-by: Arnon Kaufman <akaufman@habana.ai>

sq dirs

Change-Id: I824c6f07337ae177e2ae56213b519d8dbb098c9d
Signed-off-by: Arnon Kaufman <akaufman@habana.ai>
---

diff --git a/Makefile b/Makefile
index c2a9d8f..50e34d4 100644
--- a/Makefile
+++ b/Makefile
@@ -88,9 +88,12 @@
 	HL_VARS_DIR := $(BR2_EXTERNAL)/output
 endif
 
-$(OUTPUT_DIR):
+include scripts/make/common/hl_dirs.mk
+$(DIRS_LIST):
 	@mkdir -p $@
 
+$(OUTPUT_DIR): $(DIRS_LIST)
+
 %:
 	@$(MAKE) -C ../buildroot O=$(OUTPUT_DIR) $@
 
diff --git a/board/habanalabs/common/dirs.sh b/board/habanalabs/common/dirs.sh
new file mode 100644
index 0000000..50c46cd
--- /dev/null
+++ b/board/habanalabs/common/dirs.sh
@@ -0,0 +1,6 @@
+# NOTE: should be in sync with scripts/make/common/hl_dirs.mk
+
+export OUTPUT_DIR=${BR2_EXTERNAL}/output/${HL_DEV}/
+export IMGS_DIR=${OUTPUT_DIR}images/
+export STAGE_DIR=${IMGS_DIR}
+export BINARIES_DIR=${IMGS_DIR}
diff --git a/board/habanalabs/common/image-utility.sh b/board/habanalabs/common/image-utility.sh
index 5a0e1e7..1371775 100644
--- a/board/habanalabs/common/image-utility.sh
+++ b/board/habanalabs/common/image-utility.sh
@@ -1,5 +1,7 @@
 #!/usr/bin/env bash
 
+source ${BR2_EXTERNAL}/board/habanalabs/common/dirs.sh
+
 BINS_CONFIG_DIR="${BR2_EXTERNAL}/output/${HL_DEV}/images/bins_config"
 
 function rev_bytes_stdin {
diff --git a/board/habanalabs/fs1_arc/fs1-image-utility.sh b/board/habanalabs/fs1_arc/fs1-image-utility.sh
index 062a550..2bc2530 100755
--- a/board/habanalabs/fs1_arc/fs1-image-utility.sh
+++ b/board/habanalabs/fs1_arc/fs1-image-utility.sh
@@ -1,3 +1,5 @@
+source ${BR2_EXTERNAL}/board/habanalabs/common/dirs.sh
+
 BINS_CONFIG_DIR="${BR2_EXTERNAL}/output/${HL_DEV}/images/bins_config"
 
 function swap_file {
@@ -188,4 +190,4 @@
 
 	# Clean up
 	rm -f tmp
-}
\ No newline at end of file
+}
diff --git a/board/habanalabs/fs1_arc/fs1-post-image-common.sh b/board/habanalabs/fs1_arc/fs1-post-image-common.sh
index 2d855cd..2754c20 100755
--- a/board/habanalabs/fs1_arc/fs1-post-image-common.sh
+++ b/board/habanalabs/fs1_arc/fs1-post-image-common.sh
@@ -1,7 +1,8 @@
 #!/usr/bin/env bash
 
+source ${BR2_EXTERNAL}/board/habanalabs/common/dirs.sh
+
 SHA_VERSION=sha384
-BINARIES_DIR=${BR2_EXTERNAL}/output/${HL_DEV}/images
 BIN_IMG_DIR=${BR2_EXTERNAL}/output/${HL_DEV}_arc/images
 KEYS_DIR=${BR2_EXTERNAL}/board/habanalabs/${HL_DEV}_arc/key/
 
diff --git a/board/habanalabs/gaudi3/post-image-common.sh b/board/habanalabs/gaudi3/post-image-common.sh
index b36a3ae..4f905b3 100755
--- a/board/habanalabs/gaudi3/post-image-common.sh
+++ b/board/habanalabs/gaudi3/post-image-common.sh
@@ -1,7 +1,7 @@
 #!/usr/bin/env bash
 
+source ${BR2_EXTERNAL}/board/habanalabs/common/dirs.sh
 SHA_VERSION=sha384
-BINARIES_DIR=${BR2_EXTERNAL}/output/${HL_DEV}/images
 BIN_IMG_DIR=${BR2_EXTERNAL}/output/${HL_DEV}_arc/images
 KEYS_DIR=${BR2_EXTERNAL}/board/habanalabs/${HL_DEV}_arc/key/
 IMAGES_DIR=${BR2_EXTERNAL}/output/${HL_DEV}_arc/images
diff --git a/scripts/make/common/hl_dirs.mk b/scripts/make/common/hl_dirs.mk
new file mode 100644
index 0000000..bc4009b
--- /dev/null
+++ b/scripts/make/common/hl_dirs.mk
@@ -0,0 +1,7 @@
+# NOTE: should be in sync with board/habanalabs/common/dirs.sh
+
+export IMGS_DIR=$(OUTPUT_DIR)/images/
+export STAGE_DIR=$(IMGS_DIR)
+
+export DIRS_LIST = \
+	$(STAGING_DIR)
